src/hotspot/share/code/aotCodeCache.cpp

```diff
@@ -1993,6 +1995,8 @@ void AOTCodeCache::preload_startup_code(TRAPS) {
       if (entry->not_entrant()) {
         continue;
       }
+      Method* m = (Method*)((address)SharedBaseAddress + entry->method_offset());
```

Comment:
Why not `MetaspaceShared::requested_base_address()` here?

Reply:
Because when loading the AOT Cache, SharedBaseAddress is set to the mapped base address (in `MetaspaceShared::map_archives`). So it is fine to use it here.

Reply:
Same applies to `AOTCodeReader::read_klass` and `AOTCodeReader::read_method`.